### ISSUE TITLE
Compiler failure

### ISSUE THREAD

**Issue by kevinresol** [HaxeFoundation/haxe#9763]
4581cae5c

```haxe
Compiler failure
Please submit an issue at https://github.com/HaxeFoundation/haxe/issues/new
Attach the following information:
Haxe: 4.2.0-rc.1+4581cae5c; OS type: unix;
File "src/typing/matcher.ml", line 553, characters 11-18
Called from file "src/typing/matcher.ml", line 555, characters 10-26
Called from file "src/typing/matcher.ml", line 1641, characters 27-77
Called from file "list.ml", line 92, characters 20-23
Called from file "src/typing/matcher.ml", line 1639, characters 14-228
Called from file "src/typing/typer.ml", line 2674, characters 10-69
Called from file "src/typing/typer.ml", line 2328, characters 11-45
Called from file "src/typing/typeloadFunction.ml", line 135, characters 2-25
Called from file "std.ml", line 26, characters 6-9
Called from file "src/typing/typer.ml", line 2162, characters 17-98
Called from file "src/typing/calls.ml", line 136, characters 11-58
Called from file "src/typing/calls.ml", line 177, characters 12-33
Called from file "src/typing/calls.ml", line 190, characters 14-26
Called from file "src/typing/calls.ml", line 249, characters 15-73
Called from file "src/typing/calls.ml", line 319, characters 3-18
Called from file "src/typing/calls.ml", line 500, characters 12-62
Called from file "src/typing/matcher.ml", line 1622, characters 12-42
Called from file "src/typing/matcher.ml", line 1625, characters 19-25
Called from file "src/typing/typer.ml", line 2674, characters 10-69
Called from file "src/typing/calls.ml", line 136, characters 11-58
Called from file "src/typing/calls.ml", line 177, characters 12-33
/Users/kevin/haxe/haxe_libraries/tink_macro/0.19.3/github/328a8476a542489cbd240129cfc1fa5c0801e99e/src/tink/macro/Exprs.hx:45: characters 33-35 : Warning : Using "is" as an identifier is deprecated
src/tink/web/macros/Routing.hx:642: characters 19-21 : Warning : Using "is" as an identifier is deprecated
File "src/core/globals.ml", line 117, characters 1-7: Assertion failed
```

To reproduce:
With lix, just clone https://github.com/haxetink/tink_web/commit/c1312c28494f8493c61301365045501ac46a72d7
then `lix run travix node`

**Comment by Simn** [comment]
I'll either need a self-contained example or an ELI5 of what you want me to do with lix.

**Comment by kevinresol** [comment]
sure, just give me some time...

**Comment by kevinresol** [comment]
Let's see if this works:

1. clone https://github.com/kevinresol/tink_web/tree/issue_9763
2. install nodejs & yarn, then run `yarn` in the cloned folder
3. run `yarn haxe -js bin/node/tests.js -lib hxnodejs -lib travix -lib tink_web tests.hxml` to build
4. To build with custom haxe I think you just need to symlink it in `%AppData%/haxe/versions`

**Comment by kevinresol** [comment]
At 4898769 tink_web's test is compiling fine again. The [empty case error](https://github.com/HaxeFoundation/haxe/commit/8e7912824abafb88bcdde70ae088c9cee8c84508) never popped up. This is super weird. But anyway I think at least the problem is gone.
